### PR TITLE
Profile: disable animations on export

### PR DESCRIPTION
**Pull request by bstoeger**
### Describe the pull request:

- [x] Bug fix
- [ ] Functional change
- [ ] New feature
- [x] Code cleanup
- [ ] Build system change
- [ ] Documentation change
- [ ] Language translation

### Pull request long description:

This is my attempt at fixing #2170. It seems to have worked in my tests, but testing was very light. More testing needed.
### Changes made:

1) Store animation speed in profile object
2) Disable animations on export

### Related issues:

#2170
### Release note:

Done.
### Documentation change:

No.
### Mentions:

@atdotde

### CODE REVIEW

**Comment by atdotde** [comment]
On the phone, haven’t tried this patch. But I had the same suspicion about animations, so I inserted  1 sec of sleep between dives. This did not help.

**Comment by bstoeger** [comment]
> But I had the same suspicion about animations, so I inserted 1 sec of sleep between dives.

What kind of sleep did you use? I think you have to make sure that Qt's main loop runs. So a simple libc `sleep()` will probably not do it...

File: profile-widget/divepixmapitem.cpp [hunk 1]
```diff
@@ -108,9 +109,9 @@ void DivePictureItem::setFileUrl(const QString &s)
 
 void DivePictureItem::hoverLeaveEvent(QGraphicsSceneHoverEvent*)
 {
-	Animations::scaleTo(this, 0.2);
+	Animations::scaleTo(this, 0.2, qPrefDisplay::animation_speed());
```
Comment (atdotde) [inline profile-widget/divepixmapitem.cpp]: I think you got the order of arguments wrong (at least I get a compiler warning), should be: this, speed, 0.2

**Review by atdotde — CHANGES_REQUESTED** [review]
I can confirm this fixes the issue.

**Comment by atdotde** [comment]
I used QThread::mssleep(1000) and did not check it runs the event loop. It seems it doesn't.

File: profile-widget/divepixmapitem.cpp (hunk 1 above)
Comment (bstoeger) [reply profile-widget/divepixmapitem.cpp]: Ooops. Good catch, thanks! Will fix later.

Comment (bstoeger) [reply profile-widget/divepixmapitem.cpp]: Should be fixed now.

**Review by atdotde — APPROVED** [review]

**Pull request merged**